src/openforms/logging/service.py

```diff
@@ -0,0 +1,78 @@
+import uuid
+from dataclasses import dataclass
+from datetime import datetime
+from itertools import groupby
+from typing import Iterable
+
+from django_yubin.models import Message
+
+from openforms.logging.models import TimelineLogProxy
+from openforms.submissions.utils import get_filtered_submission_admin_url
+
+
+@dataclass
+class FailedEmail:
+    submission_uuid: uuid.UUID
+    event: str
+
+
+@dataclass
+class FailedRegistration:
+    form_uuid: uuid.UUID
+    form_name: str
+    failed_submissions_counter: int
+    initial_failure_at: datetime
+    last_failure_at: datetime
+    admin_link: str
+
+
+def collect_failed_emails(since: datetime) -> Iterable[FailedEmail]:
+    logs = TimelineLogProxy.objects.filter(
+        timestamp__gt=since,
+        extra_data__status=Message.STATUS_FAILED,
+        extra_data__include_in_daily_digest=True,
+    ).distinct("content_type", "extra_data__status", "extra_data__event")
+
+    if not logs:
+        return []
+
+    failed_emails = [
+        {
+            "submission_uuid": log.content_object.uuid,
+            "event": log.extra_data["event"],
+        }
+        for log in logs
+    ]
```

Comment:
the type signature is not respected here - you are returning dicts rather than data class instances

```suggestion
    failed_emails = [
        FailedEmail(submission_uuid=log.content_object.uuid, event=log.extra_data["event"])
        for log in logs
    ]
```